app/controllers/users_controller.rb

```diff
@@ -1,10 +1,31 @@
 class UsersController < ApplicationController
+
   def index
-    render json: Wizard.all
+    if current_user && current_user.admin?
+      @wizards = Wizard.all
+      @users = User.all
+      @reviews = Review.all
+    else
+      redirect_to wizards_path
+      flash[:notice] = "You must be signed in as an admin!"
+    end
   end
 
-  def show
-    @wizard = Wizard.find(params[:id])
-    render json: Wizard.find(params[:id]), serializer: WizardShowSerializer
+  def destroy
+    @user= User.find(params[:id])
+    if current_user.admin?
+      if @user.destroy
+        redirect_to users_path
+        flash[:notice] = "User deleted successfully"
+      else
+        flash[:notice] = 'User deletion failed'
+        render :index
+      end
+    else
+      flash[:notice] = "You must be signed in!"
```

Comment:
this routine is repeated a few times in your controllers. You could define a `protected` controller method in `ApplicationController` like:

```ruby
def deny_access(url, flash_note = "You must be signed in!")
  flash[:notice] = flash_note
  redirect_to url
end
```